### ISSUE TITLE
Segfault in lexsort. (migrated from Trac #1217)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2770]
Original ticket http://projects.scipy.org/numpy/ticket/1217
Reported 2009-09-03 by atmention:charris, assigned to unknown.

It seems to happen when sorting a reversed order array followed by an array with larger bitwise data type. Reported by Jeremy Mayes. 

```
In [1]: a = np.array(['-']*100)

In [2]: b = np.array([1.0]*100)

In [3]: i = lexsort((a,b))

In [4]: i = lexsort((a[::-1]))

In [5]: i = lexsort((b[::-1]))

In [6]: i = lexsort((a,b[::-1]))

In [7]: i = lexsort((a[::-1],b))

*Crash*

These also work:

In [3]: i = lexsort((b[::-1],a))

In [4]: i = lexsort((b[::-1],b[::-1]))

In [5]: i = lexsort((a[::-1],a[::-1]))

In [6]: i = lexsort((a,b[::-1]))
```

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2009-09-03

Should be fixed in #7356.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2009-09-03
